dotnet/src/Functions/Functions.Yaml/PromptExecutionSettingsTypeConverter.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.BufferedDeserialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Allows custom deserialization for <see cref="PromptExecutionSettings"/> from YAML prompts.
+/// </summary>
+internal sealed class PromptExecutionSettingsTypeConverter : IYamlTypeConverter
+{
+    /// <inheritdoc/>
+    public bool Accepts(Type type)
+    {
+        return type == typeof(PromptExecutionSettings);
+    }
+
+    /// <inheritdoc/>
+    public object? ReadYaml(IParser parser, Type type)
+    {
+        s_deserializer ??= new DeserializerBuilder()
+            .WithNamingConvention(UnderscoredNamingConvention.Instance)
+            .IgnoreUnmatchedProperties() // Required to ignore the 'type' property used as type discrimination. Otherwise, the "Property 'type' not found on type '{type.FullName}'" exception is thrown.
+            .WithTypeDiscriminatingNodeDeserializer(CreateAndRegisterTypeDiscriminatingNodeDeserializer)
+            .Build();
+
+        parser.MoveNext(); // Move to the first property  
+
+        var executionSettings = new PromptExecutionSettings();
+        while (parser.Current is not MappingEnd)
+        {
+            var propertyName = parser.Consume<Scalar>().Value;
+            switch (propertyName)
+            {
+                case "model_id":
+                    executionSettings.ModelId = s_deserializer.Deserialize<string>(parser);
+                    break;
+                case "function_choice_behavior":
+#pragma warning disable SKEXP0001
```

Comment:
Suggest adding this exception in the `csproj` level and removing all instances of `pragmas` for `SKEX00NN`.

```suggestion
#pragma warning disable SKEXP0001
```